packages/components/src/components/helperlabel/helperlabel.component.ts

```diff
@@ -0,0 +1,134 @@
+import { CSSResult, html, nothing } from 'lit';
+import { v4 as uuidv4 } from 'uuid';
+import { property } from 'lit/decorators.js';
+import styles from './helperlabel.styles';
+import { Component } from '../../models';
+import type { ValidationType } from './helperlabel.types';
+import { DEFAULTS, MDC_TEXT_OPTIONS } from './helperlabel.constants';
+import { getHelperIcon } from './helperlabel.utils';
+
+/**
+ * helperlabel is an internal component that contains the label and helper/validation text
+ *  that can be configured in various ways to suit different use cases (most of the input related components).
+ * It is used as an internal component and is not intended to be used directly by consumers.
+ *
+ * @tagname mdc-helperlabel
+ *
+ */
+class Helperlabel extends Component {
+/**
+     * The label of the input field. It is linked to the input field using the `for` attribute.
+     */
+@property({ reflect: true, type: String }) label = '';
+
+/**
+   * The unique id of the input field. It is used to link the input field with the label.
+   */
+@property({ type: String }) override id = `mdc-input-${uuidv4()}`;
+
+/**
+ * The type of help text. It can be 'default', 'error', 'warning', 'success', 'priority'.
+ */
+@property({ reflect: true }) helpTextType: ValidationType = DEFAULTS.VALIDATION;
+
+/**
+ * The help text that is displayed below the input field.
+ */
+@property({ type: String, reflect: true }) helpText = '';
```

Comment:
The attribute should be of `kebab-case` type
```suggestion
@property({ type: String, reflect: true, attribute: 'help-text' }) helpText = '';
```